TodoApp.ConsoleApp/Repositories/XMLRepository/TodoRepository.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Xml.Linq;
+using TodoApp.ConsoleApp.Repositories.Models;
+using TodoApp.ConsoleApp.Repositories.Interfaces;
+using System.IO;
+using System.Globalization;
+
+namespace TodoApp.ConsoleApp.Repositories.XMLRepository
+{
+    public class TodoRepository : RepositoryBase<TodoModel, Guid>, ITodoRepository
```

Comment:
In both methods there are operations that you are doing multiple times:
- Read string value from XML element;
- Read and Parse DateTime from XML element;
- Create XML element and Write string value to it;
- Create XML element and Write DateTime to it.

In order to do it consistently you can create a utility class, and move all duplicated code there.

**[This is important]** When persisting dates:
- Always save them in [ISO format](https://en.wikipedia.org/wiki/ISO_8601)!
- Always save them in UTC / GMT+0 Time zone! Do not save them in local time zone!

Example class implementation:

```cs
internal static class XmlParser
{
    public static string GetString(XElement element, string propName)
    {
        return element.Element(propName).Value;
    }

    public static DateTime GetDateTime(XElement element, string propName)
    {
        var strValue = GetString(element, propName);
        var date = DateTime.Parse(strValue, CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind);

        return date;
    }

    public static void SetString(XElement element, string propName, string value)
    {
        var propElement = new XElement(propName, value);
        element.Add(propElement);
    }

    public static void SetDateTime(XElement element, string propName, DateTime value)
    {
        var strValue = value.ToUniversalTime().ToString("o", CultureInfo.InvariantCulture);
        SetString(element, propName, strValue);
    }

    // TODO: GetGuid & SetGuid
}
```

Few notes:
- Usually creating static classes is a bad idea and not a good practice. Here we make exception, so we don't add a lot of code into the Repository class. Whenever we make exception of some rule, we have to know why we are doing it.
- We make it internal, because it is not intended to be used everywhere else. For now we have only one project, but in future we will move the Repository module in separate project. It is also possible to move XMLRepository in its own project. So we make sure that this class is not used outside XML repository.
- [`DateTime.ToUniversalTime()`](https://docs.microsoft.com/en-us/dotnet/api/system.datetime.touniversaltime?view=netframework-4.8) converts the DateTime object to UTC time zone before we save it.
- [`DateTime.ToString("o")`](https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings#Roundtrip) converts the DateTime to ISO formatted string.
- [`DateTimeStyles.RoundtripKind`](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.datetimestyles?view=netframework-4.8) parse ISO string to DateTime.

Then you can use the class in both methods:
```cs
// ElementToEntity
var entity = new TodoModel();
entity.Title = XmlParser.GetString(ContainerElement, "Title");
entity.CreatedOn = XmlParser.GetDateTime(ContainerElement, "CreatedOn");
  // etc...
return entity;

// EntityToElement
var element = new XElement("todo");
XmlParser.SetString(element, "Title", entity.Title);
XmlParser.SetDateTime(element, "CreatedOn", entity.CreatedOn);
  // etc...
return element;
```